protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```
@@ -922,7 +951,10 @@ impl ChannelFactory {
     fn update_channel(&mut self, m: &UpdateChannel) -> Option<()> {
         if let Some(channel) = self.extended_channels.get_mut(&m.channel_id) {
             let target = crate::utils::hash_rate_to_target(m.nominal_hash_rate, self.share_per_min);
-            channel.target = target;
+            match target {
+                Ok(target_) => channel.target = target_,
+                Err(e) => info!("Impossible to get target: {:?}", e),
```

this need to be an error!(..)

---

in that case the function should return an error